scripts/deploymentextensions/rundjangomigrations/install.sh

```diff
@@ -0,0 +1,224 @@
+          --target-user)
+            target_user="${arg_value}"
+            ;;          
+          --cluster-admin-email)
+            cluster_admin_email="${arg_value}"
+            ;;
+          --target-edx-system)
+            target_edx_system="${arg_value}"
+            ;;
+          --target-django-application)
+            target_django_application="${arg_value}"
+            ;;
+          --remote)
+            remote_mode=1
+            ;;
+          --debug)
+            debug_mode=1
+        esac
+
+        shift # past argument or value
+
+        if [ $shift_once -eq 0 ]; 
+        then
+            shift # past argument or value
+        fi
+
+    done
+}
+
+validate_args()
+{
+    log "Validating arguments"
+
+    # target user (only if local)
+    if [[ -z $target_user ]] && [[ $remote_mode == 0 ]];
+    then
+        log "You must specify a user account to use for SSH to remote servers"
+        exit $ERROR_DJANGO_MIGRATIONS_FAILED
+    fi
+
+    # Target server
+    if [[ -z $target_server_ip ]] ;
+    then
+        log "You must specify a server whose data directory we want to move"
+        exit $ERROR_DJANGO_MIGRATIONS_FAILED
+    fi
+
+    # cluster admin email (for notification purposes)
+    if [[ -z $cluster_admin_email ]]; 
+    then
+        log "You must specify the cluster admininstrator email address for notification purposes"
+        exit $ERROR_DJANGO_MIGRATIONS_FAILED
+    fi
+
+    if [[ -z $target_edx_system ]]; 
+    then
+        log "You must specify the edx_system as either 'lms' or 'cms'"
+        exit $ERROR_DJANGO_MIGRATIONS_FAILED
+    fi
+
+    if [[ -z $target_django_application ]]; 
+    then
+        log "You can only run migrations for a specific application. Specify django_application"
+        exit $ERROR_DJANGO_MIGRATIONS_FAILED
+    fi 
+
+    log "Completed argument validation successfully"
+}
+
+###############################################
+# START CORE EXECUTION
+###############################################
+
+# Source our utilities for logging and other base functions (we need this staged with the installer script)
+# the file needs to be first downloaded from the public repository
+current_path="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+utilities_path=$current_path/utilities.sh
+
+# check if the utilities file exists. If not, bail out.
+if [[ ! -e $utilities_path ]]; 
+then  
+    echo :"Utilities not present"
+    exit 3
+fi
+
+# source the utilities now
+source $utilities_path
+
+# Script self-identification
+print_script_header $notification_email_subject
+
+# pass existing command line arguments
+parse_args $@
+validate_args
+
+# debug mode support
+if [[ $debug_mode == 1 ]];
+then
+    set -x
+fi
+
+# sync the oxa-tools repository
+repo_url=`get_github_url "$oxa_tools_public_github_account" "$oxa_tools_public_github_projectname"`
+sync_repo $repo_url $oxa_tools_public_github_projectbranch $oxa_tools_repository_path $access_token $oxa_tools_public_github_branchtag
```

Comment:
should we pushd into a temp directory (like /tmp or /var/tmp) before doing this clone?

Reply:
Can you clarify?